### PR TITLE
[DOC] Fixing `random_state` in docstrings

### PR DESCRIPTION
**Pull request by AnonymousCodes911**
Fixes docs for `random_state`

#### Reference Issues/PRs
#1349

### CODE REVIEW

**Comment by AnonymousCodes911** [comment]
@MatthewMiddlehurst, currently this aims to fix the docstrings for only the `classification module` are these changes in line with the requirements?

File: aeon/classification/dummy.py [hunk 1]
```diff
@@ -51,10 +51,10 @@ class prior probabilities.
...
           class.
     random_state : int, RandomState instance or None, default=None
-        Controls the randomness to generate the predictions when
-        ``strategy='stratified'`` or ``strategy='uniform'``.
-        Pass an int for reproducible output across multiple function calls.
-        See :term:`Glossary <random_state>`.
+        If `int`, random_state is the seed used by the random number generator;
+        If `RandomState` instance, random_state is the random number generator;
+        If `None`, the random number generator is the `RandomState` instance used
+        by `np.random`.
```
Comment (MatthewMiddlehurst) [inline aeon/classification/dummy.py:57]: This contains some useful estimator-specific information, I don't think it would be good to remove it.

**Review by MatthewMiddlehurst — COMMENTED** [review]
Yep this is along the lines of what I expected. Be careful not to remove any useful information currently present.

File: aeon/classification/dummy.py (hunk 1 above)
Comment (AnonymousCodes911) [reply aeon/classification/dummy.py:57]: sorry for the oversight! you're correct!

**Comment by AnonymousCodes911** [comment]
@MatthewMiddlehurst, please review the additional changes made!

**Comment by MatthewMiddlehurst** [comment]
You seem to be moving outside the classification module again. Please keep it localised for now.

**Comment by AnonymousCodes911** [comment]
@MatthewMiddlehurst, I understand , 
this : [docs](https://github.com/aeon-toolkit/aeon/pull/1417/commits/b4f695d8c7c4026a492f35961f923c3d1d81215b) , wraps up the classification module. Additionally 
 the other [commit](https://github.com/aeon-toolkit/aeon/pull/1417/commits/633bb2ce25981501f848419880f861d47a6bb141), adds docstrings to the modules where only a couple of files need to be changed. Hence, I thought of clubbing them together.

**Review by wayneadams — COMMENTED** [review]
This is a great start to making these docstrings better! 

I have several comments after noticing some changes that *(I think)* were done incorrectly. Though, I'm new to this project, so I'd love someone else chime in as well.

In general, this fix is actually deceptively involved, as the `random_state` parameter can be used differently in different places. It involves following how it gets passed down into subsequent classes/methods and sometimes even external libraries.

So again, solid work on the draft and getting the conversation started.

**Comment by AnonymousCodes911** [comment]
@wayneadams , Thank you for such a detailed analysis!

File: aeon/clustering/_clara.py [hunk 2]
```diff
@@ -62,8 +62,12 @@ class TimeSeriesCLARA(BaseClusterer):
         convergence.
     verbose : bool, default=False
         Verbosity mode.
-    random_state : int or np.random.RandomState instance or None, default=None
+    random_state : int, RandomState instance or None, default=None
```
[the remark this answers is not in the document]
Comment (AnonymousCodes911) [reply aeon/clustering/_clara.py]: I think yes, `np.random.RandomState` is better!

File: aeon/forecasting/arima.py [hunk 3]
```diff
@@ -179,8 +179,12 @@ class AutoARIMA(_PmdArimaAdapter):
         True, rather than perform an exhaustive search or stepwise search,
         only n_fits ARIMA models will be fit (stepwise must be False for this
         option to do anything).
-    random_state : int, long or numpy RandomState, optional (default=None)
-        The PRNG for when random=True. Ensures replicable testing and results.
+    random_state : int, RandomState instance or None, default=None
```
[the remark this answers is not in the document]
Comment (AnonymousCodes911) [reply aeon/forecasting/arima.py]: I included those points too, but yes let's keep it consistent!

**Comment by MatthewMiddlehurst** [comment]
Looking good. Thanks @wayneadams, I think you are correct. Eventually we should go through the ones which can only take ints and see if that can be changed, but not here 🙂.

**Comment by AnonymousCodes911** [comment]
@MatthewMiddlehurst, can you please review the changes for the forecasting module in the latest commit?

**Comment by MatthewMiddlehurst** [comment]
I do not know why we needed these additional changes in this PR. I would revert them and put this up for review. I refer back to my earlier comments stating that we should do this module by module where possible.

**Comment by MatthewMiddlehurst** [comment]
you seem to have reverted too much, just the forecasting stuff..

**Comment by AnonymousCodes911** [comment]
@MatthewMiddlehurst, is this looking good now?

**Review by MatthewMiddlehurst — DISMISSED** [review]

**Review by hadifawaz1999 — DISMISSED** [review]

**Comment by MatthewMiddlehurst** [comment]
Please resolve the conflict (just accept whatever is on the main branch currently IMO)

**Review by MatthewMiddlehurst — APPROVED** [review]
LGTM, thanks.

**Pull request merged**